daniellajiddah/DeadManSwitch
# Develop

# Pull Request Details for Dead Man's Switch Project

When submitting a Pull Request (PR) to the Dead Man's Switch project, please adhere to the following guidelines:

## 1. Code Style and Quality

- Follow the existing code style in the Clarity smart contracts and TypeScript tests.
- Use meaningful and descriptive variable and function names.
- Add comments to explain complex logic or algorithms.
- Ensure your code is clean, readable, and well-organized.

## 2. Smart Contract Changes

- Any changes to `dead-mans-switch.clar` or `asset-manager.clar` should maintain the simplicity of the contracts.
- Ensure that new functions or modifications don't introduce unnecessary complexity.
- Be cautious when modifying core functionality like initialization, check-ins, or triggering the switch.

## 3. Testing

- Update or add tests in `dead-mans-switch.test.ts` for any new or modified functionality.
- Ensure all existing tests pass before submitting your PR.
- Use Vitest for writing and running tests.
- Mock contract functions and state as demonstrated in the existing tests.

## 4. Documentation

- Update the README.md file if you've added new features or changed existing ones.
- Add inline documentation for new functions or significant changes in the contracts.

## 5. Commit Messages

- Use clear and descriptive commit messages.
- Start with a verb in the present tense (e.g., "Add feature" or "Fix bug").

## 6. PR Description

- Provide a detailed description of your changes.
- Explain the problem you're solving and how your changes address it.
- If applicable, include code snippets or examples of how to use new features.

## 7. Security Considerations

- If your changes involve sensitive operations or data handling, consider potential security implications.
- Be especially careful with functions that transfer assets or modify contract state.

## 8. Performance

- Consider the impact of your changes on contract execution costs.
- Optimize for minimal storage and computation where possible.

## 9. Compatibility

- Ensure your changes are compatible with the current version of Clarity and the Stacks blockchain.
- If introducing breaking changes, clearly document the migration process.

By following these guidelines, you'll help maintain the quality and consistency of the Dead Man's Switch project. Thank you for your contribution!
